### DISCUSSION TITLE
Kneed algorithm fails for sharp data-sets

### DISCUSSION THREAD

**Discussion by DoNoRe** [arvkevi/kneed#91] [Q&A] [answered]
Hi!

I love that you put this package together..I appreciate it very much, and it works most of the time. But there are cases where it does not.

I put up a question on SO here: https://stackoverflow.com/q/61961755/11186935

Any thoughts?

**Comment by arvkevi** [comment, marked as answer]
Hi @DoNoRe, thanks for checking out the repo. Some curves have ambiguous knee points, similar to the one from your post. 

Visualizing the difference curve helps me understand why the algorithm chose the point that it did. The knee point will always be selected from a local maximum in the difference curve. In the image below I changed `S=10.0` in your example data and it moves the knee much earlier.
![knee_norm](https://user-images.githubusercontent.com/9151717/82759211-de299880-9db9-11ea-9bc1-9452950f871e.png)

It can be difficult to have the algorithm identify a "predetermined" knee point. I'd recommend tweaking a couple of parameters to try to get the knee point that you desire. Specifically `interp_method`, `online`, and `S`. You might have success fitting a line to your data, and then sending it through `kneed`.
